core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -131,6 +131,7 @@ public AggregatorParsers(ParseFieldRegistry<Aggregator.Parser> aggregationParser
                     if (token == XContentParser.Token.VALUE_STRING || token == XContentParser.Token.VALUE_EMBEDDED_OBJECT) {
                         byte[] source = parser.binaryValue();
                         binaryParser = XContentFactory.xContent(source).createParser(source);
+                        binaryParser.setParseFieldMatcher(parseContext.parseFieldMatcher());
```

Comment:
I think its okay, maybe we can add a comment to the getter that makes clear its a shortcut to the matcher in the parser.